### ISSUE TITLE
[Request] Internal methods should be fully qualified

### ISSUE THREAD

**Issue by movedoa** [Riverside-Software/sonar-openedge#296]
It would be nice to have a rule that checks if a call of an internal method is made with "THIS-OBJECT" or the class name for static methods

Wrong:
FooMethod()

Right:
THIS-OBJECT:FooMethod() or FooClass:FooMethod()

**Comment by gquerret** [comment]
Why do you want to have `THIS-OBJECT` in front of the method call ? Doesn't it just add unnecessary keywords to the code ?
But BTW, there could be two different rules, one enforcing usage of this-object, and the other one rejecting usage of this-object ; depending on coding style, the former or the latter could be activated.

**Comment by movedoa** [comment]
I think it is cleaner and more readable. You instantly see that is is an internal method call.

**Comment by jurjendijkstra-vma** [comment]
yeah but only if you also enforce "in this-procedure' when you run an internal procedure  ;-)

`run internalprocedure in this-procedure.`

ok just kidding, couldn't resist to reply while waiting for a loooong ant job

**Comment by movedoa** [comment]
I try to refactor everything to OO so this is not really a concern of mine ;-)
